tools/ci_build/build.py

```diff
@@ -1134,6 +1137,51 @@ def run_training_python_frontend_e2e_tests(cwd):
         'BertModelTest.test_for_pretraining_mixed_precision_with_gradient_accumulation'], cwd=cwd)
 
 
+def run_training_pipeline_e2e_tests(cwd):
```

Comment:
adding these to build.py is one way to do it. however, i think there's already a lot here. it's also possible to run tests from separate scripts, as in https://github.com/microsoft/onnxruntime/blob/master/tools/ci_build/github/azure-pipelines/orttraining-linux-gpu-e2e-test-ci-pipeline.yml

Reply:
I'd prefer using build.py as it has some mpi config code I can re-use in my model parallelism test. This file also contains other distributed test so kind of keeping the test centralized.